### DISCUSSION TITLE
Remove old media and directories on move to another location.

### DISCUSSION THREAD

**Discussion by ap-coder** [spatie/laravel-medialibrary#2639] [Ideas]
Tweak syncMediaPath to say moveToCloud. 

You should make a callable function similar to this but for moving files from old local/public disk to new cloud/s3 

**_I have a custom sync system created. But I am missing  a way to remove old media and directories after successful move. This would be really helpful for updating to cloud integration on existing projects._** 

```
$this->filesystem->disk($media->disk)
            ->move($factory->getPath($oldMedia), $factory->getPath($media))
            **->deleteAfterMove($media);**
```

## If you have a way to remove the old directories but not affect the new moved ones please let me know. 

```
public function syncMediaPath(Media $media): void
    {
        $factory = PathGeneratorFactory::create();

        $oldMedia = (clone $media)->fill($media->getOriginal());

        if ($oldMedia->getPath() === $media->getPath()) {
            return;
        }

        $this->filesystem->disk($media->disk)
            ->move($factory->getPath($oldMedia), $factory->getPath($media));

Missing remove old media and directories after successful move. 
This would be really helpful for updating to cloud integration on existing projects.    
}
```

**Comment by ap-coder** [comment]
I have additional info to add. 

After digging into this process i have noticed that when one is successfully moved to s3 it leaves the directory but also created a new record and leaves the old one in the DB. Is there a way to replace the old one instead of having both records in it.

**Comment by heidilux** [comment]
I'm currently migrating from local storage to a DO space as well. I used the `move()` method.
In the docs page for Moving Media, I'm just using here the media's original model as the "new" model destination, but on a new disk.

```
$media = \Spatie\MediaLibrary\MediaCollections\Models\Media::all();
foreach ($media as $item) {
    if ($model = $item->model()->first()) {
        $item->move($model, $item->collection_name, 'digitalocean');
    )
}
```

edit:
I guess it's different for me, as the files were already managed via medialibrary.
